flocker/node/agents/cinder.py

```diff
@@ -189,59 +284,70 @@ def _is_cluster_volume(cluster_id, cinder_volume):
     return False
 
 
-def _blockdevicevolume_from_cinder_volume(cinder_volume):
+def _blockdevicevolume_from_cinder_volume(cinder_volume, host_map):
     """
     :param Volume cinder_volume: The ``cinderclient.v1.volumes.Volume`` to
         convert.
     :returns: A ``BlockDeviceVolume`` based on values found in the supplied
         cinder Volume.
     """
+    if cinder_volume.attachments:
+        # There should only be one.
+        [attachment_info] = cinder_volume.attachments
+        # Nova and Cinder APIs return ID strings. Convert to UUID
+        # before looking up.
+        server_id = UUID(attachment_info['server_id'])
+        flocker_host = host_map[server_id]
+    else:
+        flocker_host = None
+    
     return BlockDeviceVolume(
         blockdevice_id=unicode(cinder_volume.id),
         size=int(GB(cinder_volume.size).to_Byte().value),
-        host=None,
+        host=flocker_host,
         dataset_id=UUID(cinder_volume.metadata[DATASET_ID_LABEL])
     )
 
 
-def rackspace_cinder_client(**kwargs):
+def rackspace_session(**kwargs):
```

Comment:
The same session can be supplied to both Nova and Cinder clients, so I updated this factory function.